src/pages/profiles/modification/parameter-selection.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useEffect, useState } from 'react';
+import HighlightOffIcon from '@mui/icons-material/HighlightOff';
+import FolderIcon from '@mui/icons-material/Folder';
+import { Grid, IconButton, Tooltip, Typography, useTheme } from '@mui/material';
+import { useIntl } from 'react-intl';
+import { DirectoryItemSelector } from '@gridsuite/commons-ui';
+import { useController, useWatch } from 'react-hook-form';
+import {
+    fetchDirectoryContent,
+    fetchPath,
+    fetchRootFolders,
+} from 'services/directory';
+import { fetchElementsInfos } from 'services/explore';
+
+// TODO remove ElementType when available in commons-ui (available soon)
+export enum ElementType {
+    DIRECTORY = 'DIRECTORY',
+    STUDY = 'STUDY',
+    FILTER = 'FILTER',
+    CONTINGENCY_LIST = 'CONTINGENCY_LIST',
+    MODIFICATION = 'MODIFICATION',
+    CASE = 'CASE',
+    VOLTAGE_INIT_PARAMETERS = 'VOLTAGE_INIT_PARAMETERS',
+    SECURITY_ANALYSIS_PARAMETERS = 'SECURITY_ANALYSIS_PARAMETERS',
+    LOADFLOW_PARAMETERS = 'LOADFLOW_PARAMETERS',
+    SENSITIVITY_PARAMETERS = 'SENSITIVITY_PARAMETERS',
+}
+
+export interface ModifyElementSelectionProps {
+    elementType: ElementType;
+    parameterFormId: string;
+    parameterNameKey: string;
+}
+
+const ParameterSelection: React.FunctionComponent<
+    ModifyElementSelectionProps
+> = (props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+
+    const [open, setOpen] = useState<boolean>(false);
+    const [selectedElementName, setSelectedElementName] = useState<string>();
+    const [validity, setValidity] = useState<boolean>();
+    const watchParamId = useWatch({
+        name: props.parameterFormId,
+    });
+    const ctlParamId = useController({
+        name: props.parameterFormId,
+    });
+
+    useEffect(() => {
+        if (!watchParamId) {
+            setSelectedElementName(undefined);
+            setValidity(undefined);
+        } else {
+            fetchPath(watchParamId)
```

Comment:
This method is called twice after clicking on few profiles

Reply:
yes, I can't get rid of it !